File: 백준/Bronze/5698. Tautogram/Tautogram.py
import sys
input = lambda: sys.stdin.readline().rstrip()


def main():
    while 1:
        S = input()
        if S == '*':
            break
        lst = list(S.split())
        a = ''
        flag = True
        for l in lst:
            if a == '' and a != l[0].upper():
                a = l[0].upper()
            elif a != l[0].upper():
                flag = False
                break
        if flag:
            print('Y')
        else:
            print('N')


if __name__ == '__main__':
    main()

File: 백준/Bronze/5698. Tautogram/Tautogram.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>

#define MAX_LENGTH 1002

int main() {
    char S[MAX_LENGTH];

    while (1) {
        fgets(S, MAX_LENGTH, stdin);
        S[strcspn(S, "\n")] = '\0';
        if (strcmp(S, "*") == 0) {
            break;
        }

        char *lst = strtok(S, " ");
        char a = '\0';
        int flag = 1;

        while (lst != NULL) {
            if (a == '\0') {
                a = toupper(lst[0]);
            } else if (a != toupper(lst[0])) {
                flag = 0;
                break;
            }
            lst = strtok(NULL, " ");
        }
        if (flag) {
            printf("Y\n");
        } else {
            printf("N\n");
        }
    }
    return 0;
}